### PR TITLE
Bugfixes

### PR DESCRIPTION
**Pull request by grechaw**
I've addressed two more bugs in this PR

#298 is all set with TDE -- I tested by modifying an existing model to add hyphens.
#328 this looks better now.  I also added a test for this issue.  instance-json-from-document was also broken.

### CODE REVIEW

**Comment by grechaw** [comment]
Updated this PR with one more commit, I'd missed a test case update.

**Comment by bsrikan** [comment]
Thanks @grechaw. I will look into this.

File: entity-services/src/main/xdmp/entity-services/entity-services-codegen.xqy [hunk 1]
```diff
@@ -289,43 +289,45 @@ declare function {$prefix}:canonicalize(
...
                     return
-                        for $val in $instance-property
-                        return map:put($m, $key, $val)
-                default return map:put($m, $key, $instance-property)
-        return $m)
-
+                        typeswitch ($instance-property)
+                        (: This branch handles embedded objects.  You can choose to prune
+                           an entity's representation of extend it with lookups here. :)
+                        case json:object
```
Comment (bsrikan) [inline entity-services/src/main/xdmp/entity-services/entity-services-codegen.xqy]: earlier it was case json:object+
Can you please let me know if removing + was intentional.

**Comment by bsrikan** [comment]
instance-to-json-envelope() used to output `"attachments":"......"` earlier.
Now in this PR its like `"attachments":[ "....." ]`

Is this expected?

File: entity-services/src/main/xdmp/entity-services/entity-services-codegen.xqy (hunk 1 above)
Comment (grechaw) [inline entity-services/src/main/xdmp/entity-services/entity-services-codegen.xqy]: Yes, I meant to do that, reasoning that within an instance, there are no sequences, only arrays as containers.  However, your pointing out my intention makes me realize that the 'for' statement just afterwards is not needed, if indeed there is just one item in $instance-property.  I'll update that code and check my unit tests.

File: entity-services/src/main/xdmp/entity-services/entity-services.xqy [hunk 2]
```diff
@@ -347,6 +346,7 @@ declare function es:add-attachments(
     $source as item()*
 ) as map:map
 {
+    let $_ := xdmp:log( ( "BAH", $source) ) return
```
Comment (grechaw) [inline entity-services/src/main/xdmp/entity-services/entity-services.xqy]: would not want to leave this in here either.

**Comment by grechaw** [comment]
Yes, attachments should be an array, as it can contain multiple things.  It was an oversight that I kept it a single value from my MLW prototype.

**Comment by grechaw** [comment]
I've added two commits which correct the sequence error, remove log stmt, and correct key.

**Pull request merged**